chromium_src/third_party/boringssl/src/crypto/curve25519/curve25519.c

```diff
@@ -0,0 +1,80 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "third_party/boringssl/src/include/openssl/curve25519.h"
+
+#include "src/third_party/boringssl/src/crypto/curve25519/curve25519.c"
+
+#ifdef UNSAFE_BUFFERS_BUILD
+#pragma allow_unsafe_buffers
```

Comment:
is this necessary? I assume `src/third_party/boringssl/src/crypto/curve25519/curve25519.c` either has this already or is built without unsafe buffers feature.

Reply:
These lines trigger error otherwise:
`(scalar[31] & 0b11000000) == 0b01000000;`
`sc_muladd(out_sig + 32, hram, scalar, nonce);`
Chromium turns off checks for almost whole third_party
https://source.chromium.org/chromium/chromium/src/+/main:build/config/unsafe_buffers_paths.txt;l=67-68 but it is enabled for brave/chromium_src/

Reply:
if this is a new file, please do not introduce unsafe buffers.

Reply:
I wanted this file to be consistent with chromium one so didn't split `out_sig` into two parts and didn't add tempting refactoring.

Alos it seems to me
`brave/chromium_src/third_party/boringssl/`
`src/third_party/boringssl/`
and in general 
`brave/chromium_src/something`
`src/something`
the first ones should follow upstream for unsafe buffer checks policy.

Reply:
I don't think consistency is an issue here, as most things are being migrated gradually, however I'm now wondering `//third_party/boringssl` can depend on `//base`. Better leave it as you did it, as this is shadowing a `.c` file for now anyway.